files/build_templates/docker_image_ctl.j2

```diff
@@ -130,8 +130,8 @@ start() {
     {%- else %}
     # Obtain our HWSKU as we will mount directories with these names in each docker
     HWSKU=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`
+    HOSTNAME=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hostname"]'`
```

Comment:
What is the reason that you have moved this line to else section?

Reply:
when you start the database, you cannot use -d opton since it requires the database to be up.